lms/djangoapps/courseware/tests/test_middleware.py

```diff
@@ -26,32 +25,24 @@ def setUpClass(cls):
     def setUp(self):
         super(CoursewareMiddlewareTestCase, self).setUp()
 
-    def check_user_not_enrolled_redirect(self):
-        """A UserNotEnrolled exception should trigger a redirect"""
-        request = RequestFactory().get("dummy_url")
-        response = RedirectUnenrolledMiddleware().process_exception(
-            request, courses.UserNotEnrolled(self.course.id)
-        )
-        self.assertEqual(response.status_code, 302)
-        # make sure we redirect to the course about page
-        expected_url = reverse(
-            "about_course", args=[self.course.id.to_deprecated_string()]
-        )
-
-        target_url = response._headers['location'][1]
-        self.assertTrue(target_url.endswith(expected_url))
-
-    def test_user_not_enrolled_redirect(self):
-        self.check_user_not_enrolled_redirect()
-
-    @patch.dict("django.conf.settings.FEATURES", {"ENABLE_MKTG_SITE": True})
```

Comment:
Is the reverse of "course_about" different if you have the marketing site enabled or not, and is this test ensuring it will work in either case?